common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/InstantRange.scala

```diff
@@ -0,0 +1,39 @@
+package uk.ac.wellcome.models.work.internal
+
+import java.time.{Instant, LocalDate, LocalDateTime, ZoneOffset}
+import uk.ac.wellcome.models.parse.Parser
+
+case class InstantRange(from: Instant,
+                        to: Instant,
+                        label: String = "",
+                        inferred: Boolean = false) {
+
+  def withInferred(inferred: Boolean): InstantRange =
+    InstantRange(from, to, label, inferred)
+
+  def withLabel(label: String): InstantRange =
+    InstantRange(from, to, label, inferred)
+}
+
+object InstantRange {
+
+  def apply(from: LocalDate, to: LocalDate, label: String): InstantRange =
+    InstantRange(
+      from.atStartOfDay(),
+      to.atStartOfDay().plusDays(1).minusNanos(1),
+      label
+    )
+
+  def apply(from: LocalDateTime,
+            to: LocalDateTime,
+            label: String): InstantRange =
+    InstantRange(
+      from.toInstant(ZoneOffset.UTC),
+      to.toInstant(ZoneOffset.UTC),
+      label
+    )
+
+  def parse(label: String)(
+    implicit parser: Parser[InstantRange]): Option[InstantRange] =
+    parser(label)
+}
```

Comment:
Meta-comment: given the overall size of this PR, it'd be helpful if changes like this (e1d30dd, splitting out InstantRange, which are much smaller and simpler to review) were split out into a separate PR.